addons/account_payment/controllers/portal.py

```diff
@@ -15,35 +18,106 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
             # Do not compute payment-related stuff if given invoice doesn't have to be paid.
             return values
 
+        common_view_values = self._get_common_page_view_values(
+            invoices_data={
+                'partner': invoice.partner_id,
+                'company': invoice.company_id,
+                'total_amount': invoice.amount_total,
+                'currency': invoice.currency_id,
+                'residual_amount': invoice.amount_residual,
+                'invoice_ids': [invoice.id],
+                'landing_route': invoice.get_portal_url(),
+            },
+            access_token=access_token,
+            **kwargs)
+        values |= common_view_values
+        return values
+
+    @http.route(['/my/invoices/overdue'], type='http', auth='public', methods=['GET'], website=True)
+    def portal_my_overdue_invoices(self, access_token=None, **kw):
+        try:
+            request.env['account.move'].check_access_rights('read')
+        except (AccessError, MissingError):
+            return request.redirect('/my')
+
+        overdue_invoices = request.env['account.move'].search(self._get_overdue_invoices_domain())
+
+        values = self._overdue_invoices_get_page_view_values(overdue_invoices, access_token, **kw)
+        return request.render("account_payment.portal_overdue_invoices_page", values) if 'payment' in values else request.redirect('/my/invoices')
+
+    def _overdue_invoices_get_page_view_values(self, overdue_invoices, access_token, **kwargs):
+        values = {'page_name': 'overdue_invoices'}
+
+        if len(overdue_invoices) == 0:
+            return values
+
+        # There is an assumption that all invoices have the same partner, company, and currency
+        # It is also assumed that they all have the same access rights
+        first_invoice = overdue_invoices[0]
+        partner = first_invoice.partner_id
+        company = first_invoice.company_id
+        currency = first_invoice.currency_id
+        access_token = access_token or first_invoice._portal_ensure_token()
+
+        total_amount = sum(overdue_invoices.mapped('amount_total'))
+        residual_amount = sum(overdue_invoices.mapped('amount_residual'))
+        payment_date = fields.Date.today()
```

Comment:
variable not very useful but if you want to keep it for readability purposes i get that, maybe putting named parameter with a default value to the function could be an idea also

Reply:
I think it does not only make the code more readable, but this value is used 2 times, so it is better than calling the `today()` function twice I guess?
Moreover, I think using named parameters would enforce some default values (e.g., `None`) which would cause errors. I mean is it fine to use named parameters even though I know the default values I set would be wrong?

Reply:
yeah it's better than calling it twice, let's keep your way of doing it's more simple :smile: